### PR TITLE
Psalm phpstan levels

### PR DESCRIPTION
**Pull request by franmomu**
This raises from 1 to 6 the phpstan level and from 6 to 4 the psalm one.

### CODE REVIEW

**Review by VincentLanglet — APPROVED** [review]
Why the psalm downgrade ?

**Comment by franmomu** [comment]
> Why the psalm downgrade ?

[In Psalm level 1 is stricter that 8, so the less the better](https://psalm.dev/docs/running_psalm/error_levels/).

File: phpstan.neon.dist [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+    - vendor/phpstan/phpstan-symfony/extension.neon
 
 parameters:
-    level: 1
+    level: 6
 
     paths:
         - src
 
+    treatPhpDocTypesAsCertain: false
```
Comment (greg0ire) [inline phpstan.neon.dist:11]: Why is this needed?

File: src/Admin/Extension/AbstractTranslatableAdminExtension.php [hunk 2]
```diff
@@ -84,6 +84,8 @@ public function getTranslatableChecker()
      * Return current translatable locale
      * ie: the locale used to load object translations != current request locale.
      *
+     * @phpstan-param AdminInterface<object> $admin
```
Comment (greg0ire) [inline src/Admin/Extension/AbstractTranslatableAdminExtension.php:87]: Should it be
```suggestion
     * @phpstan-param AdminInterface<TranslatableInterface> $admin
```

?

File: src/DependencyInjection/Compiler/AdminExtensionCompilerPass.php [hunk 3]
```diff
@@ -53,6 +56,8 @@ public function process(ContainerBuilder $container)
     }
 
     /**
+     * @param array<int, int|string> $types
```
Comment (greg0ire) [inline src/DependencyInjection/Compiler/AdminExtensionCompilerPass.php:59]: The index does not matter, I'm not sure what the best way to express this is…
I'd say either `list<int|string>` to tell people they should not have keys, or `array<int|string>` to tell them they can do whatever they want with the keys.

File: src/DependencyInjection/SonataTranslationExtension.php [hunk 4]
```diff
@@ -102,6 +106,8 @@ public function load(array $configs, ContainerBuilder $container)
     /**
      * @param array $translationTargets
      *
+     * @phpstan-param array<string, array<string, class-string[]>> $translationTargets
```
Comment (greg0ire) [inline src/DependencyInjection/SonataTranslationExtension.php:109]: That variable is only ever used in a foreach, and the key is not used…
```suggestion
     * @phpstan-param iterable<string, array<string, class-string[]>> $translationTargets
```

or even
```suggestion
     * @phpstan-param iterable<array<string, class-string[]>> $translationTargets
```

File: phpstan.neon.dist (hunk 1 above)
Comment (VincentLanglet) [reply phpstan.neon.dist:11]: Oh I missed this ! I think we should add this indeed !

Comment (greg0ire) [reply phpstan.neon.dist:11]: Why?

Comment (VincentLanglet) [reply phpstan.neon.dist:11]: Sorry. I meant "shouldnt".

Comment (VincentLanglet) [reply phpstan.neon.dist:11]: We're not supposed to check ourself if the value provided is respecting the phpdoc IMHO.

Comment (franmomu) [reply phpstan.neon.dist]: yes, you are right, I added it at the beginning of this PR and then I forgot to remove it and check. The fail was related to the constructor of `LocaleSwitcherBlockService`.

File: src/Admin/Extension/AbstractTranslatableAdminExtension.php (hunk 2 above)
Comment (franmomu) [reply src/Admin/Extension/AbstractTranslatableAdminExtension.php:87]: If I do this I get:
```
------ -------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   Admin/Extension/AbstractTranslatableAdminExtension.php
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------
  109    Parameter #1 $admin of method Sonata\TranslationBundle\Admin\Extension\AbstractTranslatableAdminExtension::getTranslatableLocale() expects
         Sonata\AdminBundle\Admin\AdminInterface<Sonata\TranslationBundle\Model\TranslatableInterface>, Sonata\AdminBundle\Admin\AdminInterface<object> given.
  115    Parameter #1 $admin of method Sonata\TranslationBundle\Admin\Extension\AbstractTranslatableAdminExtension::getTranslatableLocale() expects
         Sonata\AdminBundle\Admin\AdminInterface<Sonata\TranslationBundle\Model\TranslatableInterface>, Sonata\AdminBundle\Admin\AdminInterface<object> given.
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------
```

In line 109 we have:

https://github.com/sonata-project/SonataTranslationBundle/blob/5a93fc6176739a03f4481930d3cb05a341a2917c/src/Admin/Extension/AbstractTranslatableAdminExtension.php#L106-L109

This method is defined in [`AdminExtensionInterface`](https://github.com/sonata-project/SonataAdminBundle/blob/f2dba48d52ecd60186a7f4f530131f9fd4e1ed62/src/Admin/AdminExtensionInterface.php#L142-L149) as:

```php
/**
 * Get a chance to add persistent parameters.
 *
 * @return array<string, mixed>
 *
 * @phpstan-param AdminInterface<object> $admin
 */
public function getPersistentParameters(AdminInterface $admin);
```

I think the problem seems to be related to https://github.com/sonata-project/SonataAdminBundle/pull/6576, because looks like you cannot substitute `object` by `TranslatableInterface`.

**Review by greg0ire — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks !
